packages/cli-kit/src/public/node/ruby.ts

```diff
@@ -387,3 +389,23 @@ async function embeddedCLIExecutable(): Promise<string> {
   const cliDirectory = await shopifyCLIDirectory(true)
   return joinPath(cliDirectory, 'bin', 'shopify')
 }
+
+/**
+ * Set environment variables in case the CLI is running in a Spin environment.
+ *
+ * @returns The environment variables to set.
+ */
+async function addSpinParameters() {
+  if (!isSpinEnvironment()) return {}
+
+  const fqdn = await spinFqdn()
+  const fqdnTokens = fqdn.split('.')
+  if (fqdnTokens.length < 3) throw new AbortError(`Invalid Spin fqdn: ${fqdn}`)
```

Comment:
> In case the Spin team decide to modify the format we should adjust how the CLI should receive the spin information

While that might happen, that information can go unnoticed easily. The point that I was trying to make is that the logic is very brittle. I'd recommend exploring a more strong and explicit contract between us and the Spin CLI. For example, does `spin show --json` provide that information? If not, what if we chat with the Spin team about the idea of exposing that information too?. By doing so we have a system where the CLI contracts with Spin CLI through a very explicit contract that's way more unlikely to break.

Reply:
> Sorry but, does it make sense to invest time in making the spin team to modify their tool to adopt that cli2 specification?. I suppose that this integration will be removed once the Ruby commands are migrated to Typescript right?

I think it does, especially when we haven't brought up this use case to the Spin team and therefore we can't assess the work this would take. It might feel tempting to cargo-cult what we've done in the past or in other areas of the codebase, but we need to look at these things as opportunities to improve our code and make it more robust.

Reply:
@pepicrft as a result of our meeting we have decided to add a new env variable to the embedded `cli-ruby` that supports receiving the complete `fqdn` url